src/Dropbox.elm

```diff
@@ -720,9 +729,41 @@ type alias UploadResponse =
     }
 
 
-decodeUploadResponse : Json.Decode.Decoder UploadResponse
-decodeUploadResponse =
-    Pipeline.decode UploadResponse
+{-| Folder metadata
+-}
+type alias FolderMetadata =
+    { name : String
+    , id : String
+    , pathLower : Maybe String
+    , pathDisplay : Maybe String
+    , parentSharedFolderId : Maybe String
+    , sharedFolderId : Maybe String
+    , sharingInfo : Maybe FileSharingInfo
+    , propertyGroups : Maybe (List PropertyGroup)
+    }
+
+
+{-| Deleted item metadata
+-}
+type alias DeletedMetadata =
+    { name : String
+    , pathLower : Maybe String
+    , pathDisplay : Maybe String
+    , parentSharedFolderId : Maybe String
+    }
+
+
+{-| Metadata union type
+-}
+type Metadata
+    = FileMeta FileMetadata
+    | FolderMeta FolderMetadata
+    | DeletedMeta DeletedMetadata
```

Comment:
👍  sounds good.  I think the other option I was considering (and maybe have used somewhere already?) would be `MetadataFile` (with the name of this type being the prefix), so potentially also rename the other to `WriteConflictErrorFile`.  But I think `FileMeta` is fine for now.